app/services/curation_concerns/workflow/permission_query.rb

```diff
@@ -257,7 +257,10 @@ def scope_users_for_entity_and_roles(entity:, roles:)
             .and(entity_responsibilities[:entity_id].eq(entity.id))
         )
 
-        sub_query_for_user = agent_table.project(agent_table[:proxy_for_id]).where(
+        # PostgreSQL requires an explicit cast from string to integer
+        cast = Arel::Nodes::NamedFunction.new "CAST", [agent_table[:proxy_for_id].as("integer")]
```

Comment:
I don't think I understand what this query does, but this strikes me as a very low-level solution. Is there anything we can do at a higher level?